File: gcd.py
def gcd(a,b):
	if(b>a):
		if(b%a==0):
			print(a)
		else:
			gcd(b-a,a)
	else:
		if(a%b==0):
			print(b)
		else:
			gcd(a-b,b)
a=int(input())
b=int(input())
gcd(a,b)

File: GCD.cpp
#include<iostream>
using namespace std;
int gcd(int a,int b)
{
	if(a==b)
	{
		return b;
	}
	if(a>b)
	{
		return gcd(a-b,b);
	}
	else if(b>a)
	{
		return gcd(b-a,a);
	}
	else if(a==1)
	{
		return 1;
	}
	return -1;
}
int main() {
	int a,b;
	cin>>a>>b;
	int x=gcd(a,b);
	cout<<x;
}
